### DISCUSSION TITLE
Forwarding ref is not working as expected.

### DISCUSSION THREAD

**Discussion by x6ax6b** [solidjs/solid#1637] [Q&A] [answered]
### Describe the bug

1. Is there a way to self-reference after forwarding ref?
2. I found a way to add the ref attribute one more time and assign it to a variable, but it's probably not in the JSX spec.

### Your Example Website or App

https://playground.solidjs.com/anonymous/3eb55a2d-0390-4f04-880b-161e1db55759

### Steps to Reproduce the Bug or Issue

Please see Playground.

### Expected behavior

Hopefully this will work via props.ref, or let me know if there's a better way to do it

### Platform

- OS: macOS
- Browser: Chrome
- Version: 111.0.5563.64

**Comment by ryansolid** [comment, marked as answer]
To answer your questions:
> Problem 1: Forwarding to a parent component is great, but what about self-referencing?
> console.log("child forwarding:" + props.ref.getBoundingClientRect().height);

`props.ref` is a function so it works with the callback form. See [Forward Refs](https://www.solidjs.com/tutorial/bindings_forward_refs).

Usually I'd suggest
```js
<div ref={(el) => (selfEl = el) && props.ref?.(el)}>
  child
</div>
```
Also assigning it though is an interesting idea I hadn't thought of. That might be an interesting way to make local refs work in this case.

> Problem 2: Why does this work? (JSX elements cannot have multiple attributes with the same name.)

We don't use a standard JSX transform. So on elements technically we aren't actually merging an object and transforming each attribute. That being said this is probably unexpected behavior but I've had no reason to patch it. Other libraries using similar approaches use this to their advantage. That being said I find it is probably awkward due to the inconsistency between Components and native elements. And of course, TypeScript would never accept it.
